core/src/types/extended-repositories/profiled-repository.ts

```diff
@@ -0,0 +1,66 @@
+import type {DeepPartial, FindOneOptions} from "typeorm";
+
+import type {BaseEntity} from "../base-entity";
+import type {ExtendedRepository} from "./repository";
+
+export abstract class ProfiledRepository<T extends BaseEntity & {profile: TProfile}, TProfile extends BaseEntity> {
+    abstract readonly primaryRepository: ExtendedRepository<T>;
+
+    abstract readonly profileRepository: ExtendedRepository<TProfile>;
+
+    abstract readonly profileInverseRelationshipName: keyof TProfile;
+
+    async create(data: DeepPartial<T>): Promise<T> {
```

Comment:
Why is data a DeepPartial? Don't we need all the required fields in order to create an entity?

Reply:
Just using the type TypeORM uses for that method.

Reply:
If we don't do a partial, it'll require all relations everywhere.